# Haciendo Limpieza

In [1]:
import pandas as pd
import numpy as np
import re

In [37]:
df = pd.read_csv("data/penal_df.csv")
df.shape

(37136, 59)

Vamos a empezar limpiando las descripciones de los delitos. Tenemos una columna de código de delito y otra de descripción. Las descripciones varían, pero lo importante es que a igual código, igual delito.

In [8]:
# oju_id contiene los códigos de los delitos

cod = df['oju_id'].unique()
print(len(cod))
print(np.sort(cod))

288
[ 189  190  191  193  194  195  198  199  200  201  202  208  209  210
  211  212  213  214  215  216  217  218  219  222  223  224  225  226
  227  229  231  233  234  235  236  237  238  239  241  244  245  246
  248  252  253  567  568  569  570  571  572  573  574  576  580  581
  582  585  588  590  752  753  754  755  756  757  759  760  762  765
  766  767  768  769  772  776  778  780  783  785  797 1206 1304 1305
 1306 1307 1309 1310 1311 1312 1313 1314 1353 1368 1376 1377 1380 1384
 1385 1389 1391 1392 1393 1394 1396 1404 1411 1412 1414 1418 1420 1421
 1425 1441 1442 1443 1447 1453 1455 1458 1459 1462 1463 1467 1470 1477
 1480 1482 1485 1486 1488 1489 1501 1502 1503 1504 1508 1516 1522 1541
 1546 1547 1550 1551 1555 1561 1565 1574 1575 1577 1581 1583 1610 1611
 1612 1613 1614 1615 1616 1617 1619 1620 1621 1622 1623 1624 1625 1626
 1627 1628 1629 1630 1633 1634 1635 1636 1638 1642 1643 1645 1646 1648
 1649 1650 1651 1652 1654 1655 1656 1657 1658 1659 1661 1663 1665 1666
 1

In [32]:
# oju_descr contiene la descripción del delito.

descr = df['oju_descr'].unique()
print(len(descr))
print(descr)


287
['128 1 PARR - DELITOS ATINENTES A LA PORNOGRAFIA (PRODUCIR/PUBLICAR IMAGENES PORNOGR. C MENORES 18) CP (P/ L 2303)'
 '149 BIS  - AMENAZAS - CP (P/ L 2303)' '183 - DAÑOS'
 '181 INC. 3 - USURPACION (TURBACION DE LA POSESION) - CP (P/ L 2303)'
 '181 INC. 1 - USURPACION (DESPOJO) - CP (P/ L 2303)'
 '96  - LESIONES EN RIÑA - CP (P/ L 2303)'
 '00 - PRESUNTA COMISION DELITO (COMPETENCIA)'
 '208 INC. 1 - EJERCICIO ILEGAL DE LA MEDICINA (CURANDERISMO) - CP (P/ L 2303)'
 '189 BIS  - PORTACION DE ARMA DE FUEGO DE USO CIVIL - CP (P/ L 2303)'
 '95  - LESIONES EN RIÑA - CP (P/ L 2303)'
 '153 BIS - ACCESO SIN AUTORIZACION A UN SISTEMA O DATO INFORMATICO DE ACCESO RESTRINGIDO'
 '184 INC. 5 - DAÑOS (AGRAVADO POR EL OBJETO) - CP (P/ L 2303)'
 '129 1 PARR - EXHIBICIONES OBSCENAS' '149 BIS - AMENAZAS'
 '301BIS - JUEGOS DE AZAR SIN AUTORIZACION PERTINENTE'
 '181 INC. 1 - USURPACION (DESPOJO)'
 '189 BIS (2) - TENENCIA DE ARMA DE FUEGO DE USO CIVIL'
 '106 - ABANDONO DE PERSONAS'
 '107 - ABANDONO DE PERS

Tenemos 287 descripciones diferentes y 288 codigos. Pareceria que tenemos coincidencia casi perfecta entre el codigo y su significado.

In [34]:
descripciones = []

for row in delito:
    cod_descr = row.split(" - ")
    print(cod_descr)

['128 1 PARR', 'DELITOS ATINENTES A LA PORNOGRAFIA (PRODUCIR/PUBLICAR IMAGENES PORNOGR. C MENORES 18) CP (P/ L 2303)']
['149 BIS ', 'AMENAZAS', 'CP (P/ L 2303)']
['183', 'DAÑOS']
['181 INC. 3', 'USURPACION (TURBACION DE LA POSESION)', 'CP (P/ L 2303)']
['181 INC. 1', 'USURPACION (DESPOJO)', 'CP (P/ L 2303)']
['96 ', 'LESIONES EN RIÑA', 'CP (P/ L 2303)']
['00', 'PRESUNTA COMISION DELITO (COMPETENCIA)']
['208 INC. 1', 'EJERCICIO ILEGAL DE LA MEDICINA (CURANDERISMO)', 'CP (P/ L 2303)']
['189 BIS ', 'PORTACION DE ARMA DE FUEGO DE USO CIVIL', 'CP (P/ L 2303)']
['95 ', 'LESIONES EN RIÑA', 'CP (P/ L 2303)']
['153 BIS', 'ACCESO SIN AUTORIZACION A UN SISTEMA O DATO INFORMATICO DE ACCESO RESTRINGIDO']
['184 INC. 5', 'DAÑOS (AGRAVADO POR EL OBJETO)', 'CP (P/ L 2303)']
['129 1 PARR', 'EXHIBICIONES OBSCENAS']
['149 BIS', 'AMENAZAS']
['301BIS', 'JUEGOS DE AZAR SIN AUTORIZACION PERTINENTE']
['181 INC. 1', 'USURPACION (DESPOJO)']
['189 BIS (2)', 'TENENCIA DE ARMA DE FUEGO DE USO CIVIL']
['106', 'ABAND

In [ ]:
for num in cod:
    descr = df[df['oju_descr'].str.contains(cod)]
    print(cod)
    print(descr)

In [25]:
df.groupby(['oju_id'])['oju_descr'].count().sort_values(ascending=False)

oju_id
580     6131
211     5108
217     3241
570     2800
1391    1989
        ... 
769        1
1652       1
1651       1
1630       1
1665       1
Name: oju_descr, Length: 288, dtype: int64

In [28]:
print(df[(df['oju_id'] == 1665)].oju_descr.unique())

['277 1 / - AC. E) ENCUBRIMIENTO POR AYUDAR A ASEGURAR EL PRODUCTO DEL DELITO']


Que tiempo maximo de pena corresponde a cada delito?